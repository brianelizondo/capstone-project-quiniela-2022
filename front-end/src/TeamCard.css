.TeamCard-card a{
    text-decoration: none;
}   
.TeamCard-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 40px;
    float: left;
    width: 180px;
    height: 220px;
    margin-left: 10px;
    margin-right: 10px;
    display: flex;
    align-items: center;
}
.TeamCard-teamlogo{
    aspect-ratio: auto;
    max-height: 160px;
    max-width: 150px;
}
.TeamCard-teamname{
    color: black !important;
    width: 100% !important;
    text-align: center;
    text-transform: uppercase;
    font-size: 12px;
    font-weight: 700;
    padding-top: 10px;
}