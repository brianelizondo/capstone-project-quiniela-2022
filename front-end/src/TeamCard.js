import React from "react";
import { Link } from "react-router-dom";
import { Card } from 'react-bootstrap';

function TeamCard({ team }){        
    return (
        <Link key={team.shortName} to={`/teams/${team.shortName}`}>
            <Card>
                <Card.Body>
                    <div><img src={ `https://media.api-sports.io/football/teams/${team.apiID}.png` } alt={ team.name } /></div>
                    <div>{ team.name }</div>
                </Card.Body>
            </Card>
        </Link>
    );
}

export default TeamCard;