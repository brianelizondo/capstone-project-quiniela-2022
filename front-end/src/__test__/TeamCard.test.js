import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import TeamCard from '../TeamCard';

const team = {
    id: 9,
    name: "Argentina",
    shortName: "ARG",
    apiID: 26
}

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'TeamCard' component without crashing", () => {
        render(
            <MemoryRouter>
                <TeamCard team={team} />
            </MemoryRouter>
        );
        const textFound = screen.getByText(team.name);
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'TeamCard' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <TeamCard team={team} />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});