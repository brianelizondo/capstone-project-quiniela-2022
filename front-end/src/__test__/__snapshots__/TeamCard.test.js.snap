// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'TeamCard' matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/teams/ARG"
  >
    <div
      class="card"
      data-testid="team-ARG"
    >
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-2 col"
          >
            <img
              alt="Argentina"
              class="img-fluid"
              src="/images/team_logo/ARG.png"
            />
          </div>
          <div
            class="col"
          >
            Argentina
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
